app/src/main/java/org/oppia/android/app/translation/AppLanguageResourceHandler.kt

```diff
@@ -26,7 +26,7 @@ import javax.inject.Inject
  * instance).
  */
 class AppLanguageResourceHandler @Inject constructor(
-  private val activity: AppCompatActivity,
+  private val activity: Context,
```

Comment:
Ditto  here. This seems incorrect--we cannot guarantee an activity is injected here without explicitly requesting that type (and this **must** be an activity for language switching to function correctly).

Reply:
The only reason was to be able to create robolectric tests without the burden of dealing with activities. If the test is not needed, then there is no point. I'm curious though: why does language switching does not work without activity? Can you point me to some code?

Reply:
It has to do with how the Dagger graph is set up, actually. Injecting a context makes this handler exist 1-per-app rather than 1-per-activity, so we end up in a situation where we're using resources only bound to the application. This is an issue because language switching triggers an activity recreation via:

https://github.com/oppia/oppia-android/blob/c7349c3c24f99b16c9f5115fd410fbe9c827e1d0/app/src/main/java/org/oppia/android/app/translation/AppLanguageWatcherMixin.kt#L90

And we change the activity's resources configuration dynamically based on the current language during activity creation:

https://github.com/oppia/oppia-android/blob/c7349c3c24f99b16c9f5115fd410fbe9c827e1d0/app/src/main/java/org/oppia/android/app/activity/InjectableAppCompatActivity.kt#L76

Theoretically, this can lead to using a Resources object that doesn't have the correct corresponding language. That's definitely wrong from an API perspective, but from a practical functionality perspective I'm less sure if it'd actually break in *this* specific case. Android may rely on other signals (like the default Locale object) for determining string retrieval, or perhaps it always uses the top-most activity of the process's task stack when retrieving strings. Either way, we want to keep things per-activity to increase predictability.